inference-engine/src/vpu/graph_transformer/src/stages/dynamic_shape_resolver.cpp

```diff
@@ -20,6 +21,22 @@ void FrontEnd::parseDSR(const Model& model, const ie::CNNLayerPtr& layer, const
     VPU_THROW_UNLESS(dataProducerEdge != nullptr, "Parsing layer {} of type {} failed: input with index {} (of name {}) must have a producer",
         layer->name, layer->type, 0, data->name());
 
+    const auto ngraphNode = layer->getNode();
+    VPU_THROW_UNLESS(!ngraphNode || ngraphNode->get_input_source_output(0).get_target_inputs().size() == 1,
```

Comment:
I'd say it's better to use CNN representation: both IRv7 and IRv10 are covered since nGraph is converted to CNN anyway